### DISCUSSION TITLE
Getting Started

### DISCUSSION THREAD

**Discussion by calanchinigc** [Homebrew/discussions#5018] [Getting started]
### Output of `brew config`

```shell
python3.12 -m pip install jupyter
```

### Output of `brew doctor`

```shell
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try brew install
    xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-brew-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip.
    
    If you wish to install a non-brew packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.

[notice] A new release of pip is available: 23.3.1 -> 23.3.2
[notice] To update, run: python3.12 -m pip install --upgrade pip
```

### Description of issue

After installing /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" and following the next steps to configure the path, I entered the command: brew install python@3.12
Next, I entered the command: python3.12 -m pip install jupyter but received the error: externally-managed-environment
Can you help me get unstuck?

**Comment by SMillerDev** [comment]
Use virtualenv's. It's the way Python wants you to install things in 3.11 and above.

**Comment by jwhite007** [comment]
This seems completely ridiculous. I installed Python via Homebrew so that I wouldn't break the system Python. Now I'm being told by Homebrew that I can't use the Homebrew-installed Python because I might break it. I now have to run a venv for general Python tasks even though I don't need a project-specific environment? This is just absurd.

**Reply by SMillerDev** [reply]
Maybe you can let the python developers know, they wrote https://peps.python.org/pep-0668/ that mandates this.

**Reply by quackerex** [reply]
I think this is defiantly nice feature. I found doing anything outside a venv cause some kind of problem.
> I now have to run a venv for general Python tasks even though I don't need a project-specific environment

You could create a like global venv in your home folder to use for general tasks

**Reply by jwhite007** [reply]
@SMillerDev,

I posted my original comment before much thought on the subject. I apologize for my initial display of frustration. I use pyenv extensively at work. I think it's pyenv to the rescue on my personal computers as well. I can just use pyenv to install the latest python for my general python-related tasks, and then just have that python set as my default python in my environment.
